frontend/test/acceptance/JoinSupporterSpec.scala

```diff
@@ -68,12 +68,33 @@ class JoinSupporterSpec extends FeatureSpec with Browser
       When("Users fill in delivery address details,")
       enterDetails.fillInDeliveryAddress()
 
-      And("fill in card details,")
-      enterDetails.fillInCardDetails()
+      And("click Continue")
+      enterDetails.clickContinue()
 
       And("click 'Pay' button,")
       enterDetails.pay()
 
+      Then("the Stripe Checkout iframe should display")
+      assert(enterDetails.stripeCheckoutHasLoaded())
+
+      When("the checkout iframe is present")
+      enterDetails.switchToStripe()
+
+      Then("credit card field is present")
+      assert(enterDetails.stripeCheckoutHasCC())
+
+      Then("expiry date field is present")
+      assert(enterDetails.stripeCheckoutHasExph())
+
+      Then("CVC field is present")
+      assert(enterDetails.stripeCheckoutHasCVC())
+
+      Then("submit button is present")
+      assert(enterDetails.stripeCheckoutHasSubmit())
+
+      And("they fill in credit card payment details,")
+      enterDetails.fillInCreditCardPaymentDetailsStripe()
```

Comment:
This is doing two things: filling in details and clicking a button. I would separate out these two actions in respective methods.